### ISSUE TITLE
[MPS] Chromium crashed and reloaded when running broadcasting tests on MPS backend

### ISSUE THREAD

**Issue by Christywl** [intel/webml-polyfill#318]
**Test Env:**
Chromium Version: nightly build 70.0.3503.0 (**16d3dad**)
Platform: Mac OS 10.13.6

**Expected Result:**
No crash

**Actual Result:**
Chromium crashed and reloaded when running broadcasting tests on MPS backend
```
[1520:775:1119/155931.602171:ERROR:native_view_host_mac.mm(177)] Not implemented reached in virtual void views::NativeViewHostMac::ShowWidget(int, int, int, int, int, int)
/BuildRoot/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MetalPerformanceShaders-100.6/MPSNeuralNetwork/Filters/MPSCNNBinaryKernel.mm:1116: failed assertion `[MPSCNNAdd encode...] Error invalid operation: the clipRect.origin.z(0) + clipRect.size.depth(2) > secondaryImages.count(1)'
Received signal 6
 [0x0001193989ec]
 [0x0001193988e1]
 [0x7fff59533f5a]
 [0x000000000090]
 [0x7fff592d11ae]
 [0x7fff592991ac]
 [0x7fff35a8d4dd]
 [0x7fff35bfa057]
 [0x7fff35bf72d7]
 [0x7fff35bf675a]
 [0x0001180db2c3]
 [0x0001180e1610]
 [0x0001180d9a96]
 [0x000119981d98]
 [0x000119986441]
 [0x000119985c74]
 [0x00011997fea1]
 [0x000119980551]
 [0x0001198c07b8]
 [0x0001192e8232]
 [0x000119308ddf]
 [0x0001193092a8]
 [0x00011930b46a]
 [0x0001192fb4ea]
 [0x00011930ad8f]
 [0x7fff313128f1]
 [0x7fff313cc30c]
 [0x7fff312f5350]
 [0x7fff312f47cd]
 [0x7fff312f4033]
 [0x7fff333f1e76]
 [0x00011930bacd]
 [0x00011930a8ae]
 [0x00011932d855]
 [0x00011d40a84c]
 [0x000118edfa29]
 [0x00011a752d01]
 [0x000118edebf4]
 [0x000116fcf803]
 [0x0001082b6131]
 [0x7fff59225015]
 [0x000000000008]
[end of stack trace]
[1534:775:1119/155932.904364:ERROR:gles2_cmd_decoder.cc(18059)] [.BrowserCompositor-0x7fed8b02b000]GL ERROR :GL_INVALID_OPERATION : glCreateAndConsumeTextureCHROMIUM: invalid mailbox name
[1534:775:1119/155932.904483:ERROR:gles2_cmd_decoder.cc(12532)] [.BrowserCompositor-0x7fed8b02b000]GL ERROR :GL_INVALID_VALUE : glScheduleCALayerCHROMIUM: unsupported texture format
```
<img width="1280" alt="screen shot 2018-11-20 at 10 16 54 am" src="https://user-images.githubusercontent.com/20039821/48747200-74549e00-ecad-11e8-8653-270deba33c3a.png">

**How to Reproduce:**
1. Launch chromium and enable WebML flag
2. Visit 
     - https://brucedai.github.io/nt/test/add.html?backend=mps
     - https://brucedai.github.io/nt/test/mul.html?backend=mps

**Comment by Christywl** [comment]
on the newer builds(e.g. 8b58220), the crash log changed:
```
2019-01-17 15:01:01.401 Chromium Helper[29847:490193] *** Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[__NSSingleObjectArrayI objectAtIndex:]: index 1 beyond bounds [0 .. 0]'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff4289723b __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x00007fff69b28c76 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff42958aff -[__NSSingleObjectArrayI objectAtIndex:] + 111
	3   MPSNeuralNetwork                    0x00007fff47117a1f DeadlockedBecauseTheCPUBasedWeightUpdateForThePreviousBatchOccursAtCompletionOfThisCmdBuf_TryGPUUpdateOrNewCmdBufInstead + 161778
	4   MPSNeuralNetwork                    0x00007fff471142d7 DeadlockedBecauseTheCPUBasedWeightUpdateForThePreviousBatchOccursAtCompletionOfThisCmdBuf_TryGPUUpdateOrNewCmdBufInstead + 147626
	5   MPSNeuralNetwork                    0x00007fff4711375a DeadlockedBecauseTheCPUBasedWeightUpdateForThePreviousBatchOccursAtCompletionOfThisCmdBuf_TryGPUUpdateOrNewCmdBufInstead + 144685
	6   MPSNeuralNetwork                    0x00007fff470e8fea MPSNeuralNetwork + 262122
	7   MPSNeuralNetwork                    0x00007fff470f0c3a DeadlockedBecauseTheCPUBasedWeightUpdateForThePreviousBatchOccursAtCompletionOfThisCmdBuf_TryGPUUpdateOrNewCmdBufInstead + 2573
	8   MPSNeuralNetwork                    0x00007fff470f0880 DeadlockedBecauseTheCPUBasedWeightUpdateForThePreviousBatchOccursAtCompletionOfThisCmdBuf_TryGPUUpdateOrNewCmdBufInstead + 1619
	9   MPSNeuralNetwork                    0x00007fff470ed1c9 MPSNeuralNetwork + 278985
	10  MPSNeuralNetwork                    0x00007fff470ecbc3 MPSNeuralNetwork + 277443
	11  Chromium Framework                  0x0000000111d2fb46 _ZN2ml19ExecutionImplMacMPS12StartComputeEN4base12OnceCallbackIFviEEE + 1654
	12  Chromium Framework                  0x0000000111d354e0 _ZN2ml5mojom21ExecutionStubDispatch19AcceptWithResponderEPNS0_9ExecutionEPN4mojo7MessageENSt3__110unique_ptrINS4_25MessageReceiverWithStatusENS7_14default_deleteIS9_EEEE + 224
	13  Chromium Framework                  0x0000000111d27136 _ZN2ml5mojom13ExecutionStubIN4mojo19RawPtrImplRefTraitsINS0_9ExecutionEEEE19AcceptWithResponderEPNS2_7MessageENSt3__110unique_ptrINS2_25MessageReceiverWithStatusENS9_14default_deleteISB_EEEE + 38
	14  Chromium Framework                  0x00000001135d72e8 _ZN4mojo23InterfaceEndpointClient22HandleValidatedMessageEPNS_7MessageE + 456
	15  Chromium Framework                  0x00000001135db991 _ZN4mojo8internal15MultiplexRouter22ProcessIncomingMessageEPNS1_14MessageWrapperENS1_18ClientCallBehaviorEPN4base19SequencedTaskRunnerE + 705
	16  Chromium Framework                  0x00000001135db1c4 _ZN4mojo8internal15MultiplexRouter6AcceptEPNS_7MessageE + 196
	17  Chromium Framework                  0x00000001135d53f1 _ZN4mojo9Connector17ReadSingleMessageEPj + 305
	18  Chromium Framework                  0x00000001135d5aa1 _ZN4mojo9Connector24ReadAllAvailableMessagesEv + 97
	19  Chromium Framework                  0x0000000113515d08 _ZN4mojo13SimpleWatcher13OnHandleReadyEijRKNS_18HandleSignalsStateE + 248
	20  Chromium Framework                  0x0000000112f3d782 _ZN4base5debug13TaskAnnotator7RunTaskEPKcPNS_11PendingTaskE + 242
	21  Chromium Framework                  0x0000000112f5e32f _ZN4base11MessageLoop7RunTaskEPNS_11PendingTaskE + 479
	22  Chromium Framework                  0x0000000112f5e7f8 _ZN4base11MessageLoop6DoWorkEv + 424
	23  Chromium Framework                  0x0000000112f609ba _ZN4base24MessagePumpCFRunLoopBase7RunWorkEv + 42
	24  Chromium Framework                  0x0000000112f50a3a _ZN4base3mac15CallWithEHFrameEU13block_pointerFvvE + 10
	25  Chromium Framework                  0x0000000112f602df _ZN4base24MessagePumpCFRunLoopBase13RunWorkSourceEPv + 63
	26  CoreFoundation                      0x00007fff4282f8f1 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
	27  CoreFoundation                      0x00007fff428e930c __CFRunLoopDoSource0 + 108
	28  CoreFoundation                      0x00007fff42812350 __CFRunLoopDoSources0 + 208
	29  CoreFoundation                      0x00007fff428117cd __CFRunLoopRun + 1293
	30  CoreFoundation                      0x00007fff42811033 CFRunLoopRunSpecific + 483
	31  Foundation                          0x00007fff4490ee76 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 277
	32  Chromium Framework                  0x0000000112f6101d _ZN4base20MessagePumpNSRunLoop5DoRunEPNS_11MessagePump8DelegateE + 125
	33  Chromium Framework                  0x0000000112f5fdfe _ZN4base24MessagePumpCFRunLoopBase3RunEPNS_11MessagePump8DelegateE + 110
	34  Chromium Framework                  0x0000000112f82da5 _ZN4base7RunLoop3RunEv + 53
	35  Chromium Framework                  0x0000000117060cac _ZN7content7GpuMainERKNS_18MainFunctionParamsE + 1548
	36  Chromium Framework                  0x0000000112b34f79 _ZN7content21ContentMainRunnerImpl3RunEb + 441
	37  Chromium Framework                  0x00000001143a8261 _ZN15service_manager4MainERKNS_10MainParamsE + 2849
	38  Chromium Framework                  0x0000000112b34144 _ZN7content11ContentMainERKNS_17ContentMainParamsE + 68
	39  Chromium Framework                  0x0000000110c1f903 ChromeMain + 179
	40  Chromium Helper                     0x000000010612b131 main + 497
	41  libdyld.dylib                       0x00007fff6a742015 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
Received signal 6
 [0x000112fedf3c]
 [0x000112fede31]
 [0x7fff6aa50f5a]
 [0x000000020008]
 [0x7fff6a7ee1ae]
 [0x7fff686e8f8f]
 [0x7fff686e912b]
 [0x7fff69b2aea3]
 [0x7fff687047c9]
 [0x7fff6870426f]
 [0x7fff69b28da5]
 [0x7fff42958aff]
 [0x7fff47117a1f]
 [0x7fff471142d7]
 [0x7fff4711375a]
 [0x7fff470e8fea]
 [0x7fff470f0c3a]
 [0x7fff470f0880]
 [0x7fff470ed1c9]
 [0x7fff470ecbc3]
 [0x000111d2fb46]
 [0x000111d354e0]
 [0x000111d27136]
 [0x0001135d72e8]
 [0x0001135db991]
 [0x0001135db1c4]
 [0x0001135d53f1]
 [0x0001135d5aa1]
 [0x000113515d08]
 [0x000112f3d782]
 [0x000112f5e32f]
 [0x000112f5e7f8]
 [0x000112f609ba]
 [0x000112f50a3a]
 [0x000112f602df]
 [0x7fff4282f8f1]
 [0x7fff428e930c]
 [0x7fff42812350]
 [0x7fff428117cd]
 [0x7fff42811033]
 [0x7fff4490ee76]
 [0x000112f6101d]
 [0x000112f5fdfe]
 [0x000112f82da5]
 [0x000117060cac]
 [0x000112b34f79]
 [0x0001143a8261]
 [0x000112b34144]
 [0x000110c1f903]
 [0x00010612b131]
 [0x7fff6a742015]
[end of stack trace]
[29859:775:0117/150101.913201:ERROR:gles2_cmd_decoder.cc(18059)] [.BrowserCompositor-0x7f8aa59bf800]GL ERROR :GL_INVALID_OPERATION : glCreateAndConsumeTextureCHROMIUM: invalid mailbox name
[29859:775:0117/150101.913313:ERROR:gles2_cmd_decoder.cc(12532)] [.BrowserCompositor-0x7f8aa59bf800]GL ERROR :GL_INVALID_VALUE : glScheduleCALayerCHROMIUM: unsupported texture format
[29859:775:0117/150101.913339:ERROR:gles2_cmd_decoder.cc(18059)] [.BrowserCompositor-0x7f8aa59bf800]GL ERROR :GL_INVALID_OPERATION : glCreateAndConsumeTextureCHROMIUM: invalid mailbox name
```

**Comment by fujunwei** [comment]
Maybe it is a bug for [MPSNNBinaryArithmeticNode](https://developer.apple.com/documentation/metalperformanceshaders/mpsnnbinaryarithmeticnode?language=objc)  because there is no effect when setting value to  property such as primaryStrideInPixelsX/primaryStrideInPixelsY/primaryStrideInFeatureChannels.

**Comment by Christywl** [comment]
This issue has been fixed on the nightly build 1e6cc6e.
